ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedSessionTable.java

```diff
@@ -61,28 +63,35 @@
   private List<WindowedRow> findSession(
       final ReadOnlySessionStore<Struct, GenericRow> store,
       final Struct key,
-      final Range<Instant> windowStart
+      final Range<Instant> windowStart,
+      final Range<Instant> windowEnd
   ) {
     try (KeyValueIterator<Windowed<Struct>, GenericRow> it = store.fetch(key)) {
```

Comment:
Just for my own knowledge, why don't we fetch from the store using the same bounds as in `KsMaterializedWindowTable.java`.  Is it because the window sizes are not uniform in size and therefore we don't know exactly where to stop if all we know is the window start range?  If they provide a window end range now, it seems like we could provide bounds to the store lookup (and could use Long.MAX_VALUE anyway).

Reply:
The Streams API for session tables is different.  There is no `fetch(key, lower, upper)` method.